annotations/builder/src/main/resources/snippets/validate.txt

```diff
@@ -0,0 +1,25 @@
+if (!validationEnabled) {
+  return;
+}
+Validator validator = null;
+try {
+  ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+  validator = factory.getValidator();
+} catch(ValidationException e) {
+   return;
+}
+Set<ConstraintViolation<T>> violations = validator.validate(item);
+if (!violations.isEmpty()) {
+  StringBuilder sb = new StringBuilder("Constraint Validations: ");
+  boolean first = true;
+  for (ConstraintViolation violation: violations) {
+     if (first) {
+        first = false;
+     } else {
+       sb.append(", ");
+     }
+     Object leafBean = violation.getLeafBean();
+     sb.append(violation.getPropertyPath() + " " + violation.getMessage() + " on bean: " + leafBean);
```

Comment:
thanks - looks good. I figured the generation of the message might just be a helper function; to keep the code generation simpler (and result in less bytecode)? i.e. just have a static function which generates a canonical message given an `Iterable< ConstraintViolation>`